common/lib/xmodule/xmodule/conditional_module.py

```diff
@@ -10,20 +10,68 @@
 
 from xmodule.x_module import XModule, STUDENT_VIEW
 from xmodule.seq_module import SequenceDescriptor
-from xblock.fields import Scope, ReferenceList
+from xmodule.studio_editable import StudioEditableModule, StudioEditableDescriptor
 from xmodule.modulestore.exceptions import ItemNotFoundError
+from xmodule.validation import StudioValidation, StudioValidationMessage
+from xblock.fields import Scope, ReferenceList, String
+from xblock.fragment import Fragment
 
 
 log = logging.getLogger('edx.' + __name__)
 
+# Make '_' a no-op so we can scrape strings
+_ = lambda text: text
+
 
 class ConditionalFields(object):
     has_children = True
-    show_tag_list = ReferenceList(help="List of urls of children that are references to external modules", scope=Scope.content)
-    sources_list = ReferenceList(help="List of sources upon which this module is conditional", scope=Scope.content)
-
-
-class ConditionalModule(ConditionalFields, XModule):
+    display_name = String(
+        display_name=_("Display Name"),
+        help=_("Display name for this module"),
+        scope=Scope.settings,
+        default=_('Conditional')
+    )
+
+    show_tag_list = ReferenceList(
+        help=_("List of urls of children that are references to external modules"),
+        scope=Scope.content
+    )
+
+    sources_list = ReferenceList(
+        display_name=_("Source Components"),
+        help=_("The component location(s) whose attributes are used to determine whether a student "
+               "should be shown the content inside of this conditional block."),
+        scope=Scope.content
+    )
+
+    condional_attr = String(
+        display_name=_("Conditional Attribute"),
+        help=_("The attribute from the source component used to determine whether students "
+               "should be shown the content in this conditional block."),
+        scope=Scope.content,
+        default='correct',
+        values=lambda: [{'display_name': xml_attr, 'value': xml_attr}
+                        for xml_attr in ConditionalModule.conditions_map.keys()]
+    )
+
+    conditional_value = String(
+        display_name=_("Conditional Value"),
+        help=_("The value of the conditional attribute that needs to be true for a given student "
+               "to see the content in this conditional block."),
+        scope=Scope.content,
+        default='True'
+    )
+
+    conditional_message = String(
```

Comment:
Do we need to support a way to localize the messages? It seems odd that the default message is localized, but the author's message is not.